docs/exceptions.md

```diff
@@ -217,6 +272,9 @@ two reasons for this:
 * when in an abstract (but potentially impure) monad, using `throwM` might
   add an additional constraint
 
+*Comment:* why should there be a distinction between a bug in a pure program
```

Comment:
There isn't a distinction between a bug in a pure program and in an effectful one, that's why the guidelines recommend to use the same method to represent both. But in impure code we have other techniques available (`throwM`, `throwIO`), so the document explicitly states not to use them to represent bugs, that's why we have separate sections for bugs in pure and effectful code.